### PR TITLE
possible descriptor type detection

### LINKED ISSUE #874: Extract/validate needs to guess descriptor type
# Overview

Currently, this one will not work as expected:

```python
report = validate(descriptor='datapackage.yaml') # will be handled as a default (resource)
```

### PR DESCRIPTION
**Pull request by mor120**
# Overview

closes #874 
---
It is a possible way for detecting a type by a descriptor, if the type is not provided.
As @roll suggested, this is done by creating a file for the descriptor.

Please preserve this line to notify @roll (lead of this repository)

### CODE REVIEW

File: frictionless/validate/main.py [hunk 1]
```diff
@@ -26,6 +26,11 @@ def validate(source=None, type=None, **options):
         type = "package" if file.multipart else file.type
         if type == "table":
             type = "resource"
+        if "descriptor" in options:
+            file = system.create_file(
+                options.get("descriptor"), basepath=options.get("basepath", "")
+            )
+            type = file.type
```
Comment (roll) [inline frictionless/validate/main.py]: I would fall back to `resource` type if the descriptor is detected to be a `table` as `file.type == 'table'`. Otherwise, something like this `validate(descriptor='table.csv')` will crash the lib (we might need to add a test for this edge case)

File: tests/validate/test_package.py [hunk 2]
```diff
@@ -465,3 +465,8 @@ def test_validate_package_using_detector_schema_sync_issue_847():
         resource.detector = Detector(schema_sync=True)
     report = validate(package)
     assert report.valid
+
+
+def test_validate_package_descriptor_type():
```
Comment (roll) [inline tests/validate/test_package.py]: Can you please add a few more tests like for `resource`? And some edge cases like `validate(descriptor='table.csv')`

**Review by roll — CHANGES_REQUESTED** [review]
Hi @mor120, thanks for helping with this issue!

Here are a few suggestions I added to the code and also:
- Can you please add a similar logic to `extract` function?

File: frictionless/validate/main.py (hunk 1 above)
Comment (mor120) [reply frictionless/validate/main.py]: I have added a fall back to resource if the type is not a package.

Furthermore, I would like to mention that passing a ```table``` type file as a ```descriptor``` like - ```validate(descriptor='table.csv')``` won't raise any exception but will retrieve an error: 
```Python
print(validate(descriptor='table.csv').flatten())
[1, None, None, 'resource-error']
```

**Comment by mor120** [comment]
Hey @roll,
Following the last commit I have made, I think we should add a test to  ```tests/extract/test_resource.py``` that will make sure that a table passed as a descriptor will be handled as a resource.

So that when calling the function:

```Python
extract(descriptor="data/package/data.csv")
```
A **resource-error** exception will be raised:

```Python
frictionless.exception.FrictionlessException: [resource-error].....
```

And not **general-error** exception:
```Python
frictionless.exception.FrictionlessException: [general-error] .....
```

What do you think? Is it necessary?

**Review by roll — APPROVED** [review]
Thanks @mor120!

It looks good. The `resource-errors` error I think is fine
